onnxruntime/core/mlas/lib/activate_fp16.cpp

```diff
@@ -76,9 +76,209 @@ struct MLAS_HALF_ACTIVATION_FUNCTION<MlasLeakyReluActivation>
     }
 };
 
-template<>
-struct MLAS_HALF_ACTIVATION_FUNCTION<MlasClipActivation>
-{
+//
+// MlasLogisticActivation (Sigmoid) is ported from XNNPACK
```

Comment:
you may want to guard the neon implementation using MLAS_TARGET_ARM64 or MLAS_TARGET_ARM64EC, as we will need to have Intel x64 implementations in the near future.

Reply:
or preferably you can use hardware agnostic intrinsic wrappers instead of neon intrinsic directly, e.g. the  vaddq_f16 is wrapped in inline function MlasAddFloat16x8.  When we need to add Intel x64 support we can just redefine MlasAddFloat16x8 wrapper instead of this logic

Reply:
Guard the neon implementation. Would use hardware agnostic intrinsic wrappers during adding Intel x64 support so we would know what abstraction should be at that time.